python-sdk/pachyderm_sdk/api/pps/_additions.py

```diff
@@ -0,0 +1,10 @@
+from . import CreatePipelineRequest, Service, Spout
+
+def _Create_Pipeline_Request__post_init__(self: "CreatePipelineRequest") -> None:
```

Comment:
I think the simpler approach here is to have this __post_init__ method be implemented on the `Spout` object rather than the CreatePipelineRequest object. What was the reason you chose this object?

Reply:
yeah, that was the direction i had wanted to go initially as well. but i was not able to get it working; this is what i tried:
```
def _Spout__post_init__(self: "Spout") -> None:
    if not self.service:
        self.service = Service()

Spout.__post_init__ = _Spout__post_init__
```
i feel like the spout object should get created which should cause this to kick in but maybe i'm wrong on that.